### PR TITLE
Fix offset panic

### PR DESCRIPTION
**Pull request by smellthemoon**
Signed-off-by: lixinguo <xinguo.li@zilliz.com>
#21223

### CODE REVIEW

File: internal/proxy/task_query.go [hunk 1]
```diff
@@ -492,7 +492,11 @@ func reduceRetrieveResults(ctx context.Context, retrieveResults []*internalpb.Re
 		loopEnd = int(queryParams.limit)
 
 		if queryParams.offset > 0 {
-			for i := int64(0); i < queryParams.offset; i++ {
+			loopCnt := int(queryParams.offset)
+			if loopCnt >= len(ret.FieldsData) {
```
Comment (congqixia) [inline internal/proxy/task_query.go:496]: Looks like `offset` is not related to `len(ret.FieldsData)`

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by smellthemoon** [comment]
rerun ut

**Comment by congqixia** [comment]
Tested with this solution, got this error:
`failed to query result, err: FieldData type not matched`

**Comment by smellthemoon** [comment]
rerun ut

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**